pkg/sql/catalog/internal/catkv/catalog_reader.go

```diff
@@ -133,21 +120,55 @@ func (cr catalogReader) Codec() keys.SQLCodec {
 	return cr.codec
 }
 
+// Reset is part of the CatalogReader interface.
+func (cr catalogReader) Reset(_ context.Context) {}
+
+// Cache is part of the CatalogReader interface.
+func (cr catalogReader) Cache() nstree.Catalog {
+	return nstree.Catalog{}
+}
+
+// IsIDInCache is part of the CatalogReader interface.
+func (cr catalogReader) IsIDInCache(_ descpb.ID) bool {
+	return false
+}
+
+// IsNameInCache is part of the CatalogReader interface.
+func (cr catalogReader) IsNameInCache(_ catalog.NameKey) bool {
+	return false
+}
+
+// IsDescIDKnownToNotExist is part of the CatalogReader interface.
+func (cr catalogReader) IsDescIDKnownToNotExist(_, _ descpb.ID) bool {
+	return false
+}
+
 // ScanAll is part of the CatalogReader interface.
 func (cr catalogReader) ScanAll(ctx context.Context, txn *kv.Txn) (nstree.Catalog, error) {
 	var mc nstree.MutableCatalog
-	log.Eventf(ctx, "fetching all descriptors and namespace entries")
-	cq := catalogQuery{catalogReader: cr}
+	cq := catalogQuery{codec: cr.codec}
 	err := cq.query(ctx, txn, &mc, func(codec keys.SQLCodec, b *kv.Batch) {
 		b.Header.MaxSpanRequestKeys = 0
 		descsPrefix := catalogkeys.MakeAllDescsMetadataKey(codec)
 		b.Scan(descsPrefix, descsPrefix.PrefixEnd())
+		if isEventLoggingEnabled(ctx) {
```

Comment:
Where'd you get this pattern? If you're using `VEventf` I don't think you need to wrap these in the conditional.

Reply:
The problem is that it `VEventf`ing unconditionally pollutes the event log and fails a bunch of tests which don't care about namespace lookups and other catalog operations.

Reply:
alrighty, fair enough. Another thing you could do is lift the check up so we don't need all these conditionals by assigning a closure above this function?

Like:

```go
eventf := log.VEventf
if !isEventLoggingEnabled() {
    evenf = func...
}
```